.github/workflows/ci.yml

```diff
@@ -59,9 +37,97 @@ jobs:
+      - name: Archive rhel-ubi8 Build Artifact                                                                                                                                                        
+        uses: actions/upload-artifact@v2                                                                                                                                                                      
+        # Run for main branch only                                                                                                                                                                            
+        if: github.ref == 'refs/heads/main'                                                                                                                                                                   
+        with:                                                                                                                                                                                                 
+          name: "DC.rhel.x64.${{ env.PACKAGE_NAME }}"                                                                                                                          
+          path: |                                                                                                                                                                                             
+            ./build/${{ env.PACKAGE_NAME }}
+            ./setup/
+
+  gpp-compat:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [ 5,6,7,8 ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Build ${{ env.PACKAGE_NAME }}
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u awslabs --password-stdin
+          export DOCKER_IMAGE=docker.pkg.github.com/awslabs/aws-iot-device-client/ubuntu-16-x64:latest
+          docker pull $DOCKER_IMAGE
+          docker run --mount type=bind,source=$(pwd),target=/root/${{ env.PACKAGE_NAME }} --env GITHUB_REF $DOCKER_IMAGE --compiler=g++-${{ matrix.version }}
+
+  clangpp-compat:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        version: [ 5.0, 6.0, 7, 8, 9 ]
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Build ${{ env.PACKAGE_NAME }}
+        run: |
+          echo "${{ secrets.GITHUB_TOKEN }}" | docker login docker.pkg.github.com -u awslabs --password-stdin
+          export DOCKER_IMAGE=docker.pkg.github.com/awslabs/aws-iot-device-client/ubuntu-16-x64:latest
+          docker pull $DOCKER_IMAGE
+          docker run --mount type=bind,source=$(pwd),target=/root/${{ env.PACKAGE_NAME }} --env GITHUB_REF $DOCKER_IMAGE --compiler=clang-${{ matrix.version }}
+
   build-armhf32:
     runs-on: ubuntu-latest
-    if: github.ref == 'refs/heads/main'  
```

Comment:
Why are you removing this conditional check? Won't this increase the status check times for all branches? And is there an account level limit for number of actions workflows we can run?

Reply:
There is an account level limit for the number of workflows at a given time I believe.  And the number of jobs in a workflow is limited by time, where we're no where near the cap.  I also felt that we should be building these each time for more transparency, it doesn't really add much time to wait.